ui-src/src/app/automated-ui-services.ts

```diff
@@ -8,32 +8,36 @@ import { Observable } from "rxjs/Observable";
 import { environment } from "../environments/environment.prod";
 import { $ } from "protractor";
 
-
 @Injectable()
 export class AutomatedUIServices {
-
-  constructor(private http: HttpClient) {}
-
+  private headers: HttpHeaders;
+  constructor(private http: HttpClient) {
+    this.headers = new HttpHeaders({ "Content-Type": "application/json" });
+  }
 
   postConfig$(configData, configPath): Observable<any> {
-    let headers = new HttpHeaders();
-    headers = headers.set("Content-Type", "application/json;");
-    headers = headers.append("configPathDir", configPath);
-    return this.http.post(`${environment.apiBaseUrl}configs`,configData, {headers});
-
+    const headers = this.headers.append("configPathDir", configPath);
+    return this.http.post(`${environment.apiBaseUrl}configs`, configData, {
+      headers: headers
+    });
   }
   getConfig$(configPath, xmlFileName): Observable<any> {
-    let headers = new HttpHeaders();
-    headers = headers.set("Content-Type", "application/json;");
-    headers = headers.append("configPathDir", configPath);
-    return this.http.get(`${environment.apiBaseUrl}configs/${xmlFileName}`, {headers});
+    const headers = this.headers.append("configPathDir", configPath);
```

Comment:
There is actually no need for the const. You can simply do

```
this.headers = this.headers.append("configPathDir", configPath);

return this.http.get(`${environment.apiBaseUrl}configs/${xmlFileName}`, {headers: this.headers});
```

Reply:
it needs to be set like this because it wont be treated as a new instance and the 2nd time you try a GET for example the append will add the configPath again causing the path to be : C:\\U
sers\\A586754\\go\\src\\github.com\\xtracdev\\automated-perf-test\\config,C:\\Us
ers\\A586754\\go\\src\\github.com\\xtracdev\\automated-perf-test\\config/config.
xml"

Reply:
ah ok, well then what about changing to use set instead of append.